bleak/backends/winrt/client.py

```diff
@@ -452,27 +453,32 @@ async def unpair(self) -> bool:
             Boolean on whether the unparing was successful.
 
         """
+        if self._requester:
+            device = self._requester
+        else:
+            device = await BluetoothLEDevice.from_bluetooth_address_async(
+                address_to_int(self.address)
+            )
 
         # New local device information object created since the object from the requester isn't updated
         device_information = await DeviceInformation.create_from_id_async(
-            self._requester.device_information.id
+            device.device_information.id
         )
-        if device_information.pairing.is_paired:
-            unpairing_result = await device_information.pairing.unpair_async()
 
-            if unpairing_result.status not in (
-                DeviceUnpairingResultStatus.UNPAIRED,
-                DeviceUnpairingResultStatus.ALREADY_UNPAIRED,
-            ):
-                raise BleakError(
-                    f"Could not unpair with device: {unpairing_result.status}"
-                )
+        unpairing_result = await device_information.pairing.unpair_async()
 
-            else:
-                logger.info("Unpaired with device.")
-                return True
+        if not self._requester:
+            # close the device object if this was created locally
+            device.close()
```

Comment:
It is unlikely but still theoretically possible that `self._requester` could be assigned while awaiting, so perhaps we should just have a local `should_close` flag instead?

Reply:
I added a try/finally block to ensure close() is called in case of an error. I also added the should_close flag.
Is it even necessary to use self._requester here at all? Can we just create a new BluetoothLEDevice for every call to unpair an simplify this function?
Something like this:
```
        device = await BluetoothLEDevice.from_bluetooth_address_async(
            _address_to_int(self.address)
        )

        try:
            unpairing_result = await device.device_information.pairing.unpair_async()

        finally:
            device.close()

        if unpairing_result.status not in (
            DeviceUnpairingResultStatus.UNPAIRED,
            DeviceUnpairingResultStatus.ALREADY_UNPAIRED,
        ):
            raise BleakError(f"Could not unpair with device: {unpairing_result.status}")

        logger.info("Unpaired with device.")
        return True
```
What do you think?

Reply:
Not all devices have a static address. Bonded devices with privacy enabled will have a random address that changes from time to time.  So `from_bluetooth_address_async()` should only be called as a last resort if no other information is available.

In fact, we may already have `self._device_info` even if `self_requester` is None. So we could do something like this:

```python
if self._device_info is None:
    # BleakClient was initialized with address instead of BleDevice and has not connected yet
    device = await BluetoothLEDevice.from_bluetooth_address_async(
        _address_to_int(self.address)
    )
    device_information = device.device_information
    device.close()
else:
    device_information = self._device_info

...
```

Reply:
``self._device_info`` does only contain the Bluetooth address from advertising. It is no ``DeviceInformation`` object which can be used for unpairing.
``self._device_info = device.details.adv.bluetooth_address``

I updated the PR and added a _create_requester function which takes the address type into account and can be used by ``connect`` and ``unpair`` to create the ``BluetoothLEDevice`` object. 
For ``unpair`` the address from self._device_info is used if it is available (as it was already done in ``connect`` before). Otherwise self.address is used.
I also recognized that ``from_bluetooth_address_async`` can return None if the device is not paired and not found in the system cache. So I added a check for this case.